headers.go

```diff
@@ -211,6 +214,9 @@ func (h *UnprotectedHeader) UnmarshalCBOR(data []byte) error {
 	if err := decMode.Unmarshal(data, &header); err != nil {
 		return err
 	}
+	if err := ensureNoCritical(header); err != nil {
+		return err
```

Comment:
see also suggestion below

```suggestion
		return fmt.Errorf("unprotected header: %w")
```